morerss/zhihu.py

```diff
@@ -121,52 +118,48 @@ async def get(self, name):
 
   async def _get_url(self, url):
     res = await zhihulib.fetch_zhihu(url)
-    info = json.loads(res.body.decode('utf-8'))
-    return info
+    return json.loads(res.body.decode('utf-8'))
 
 def post2rss(baseurl, post, *, digest=False, pic=None, fullonly=False):
   url = post['url']
   if digest:
     content = post['excerpt']
     content = zhihulib.process_content_for_html(content, pic=pic)
+  elif article := article_from_cache(post['id'], post['updated']):
+    # logger.debug('cache hit for %s', post['id'])
+    base.STATSC.incr('zhihu.cache_hit')
+
+    content = article['content']
+    if pic:
+      doc = fromstring(content)
+      base.proxify_pic(doc, zhihulib.re_zhihu_img, pic)
+      content = tostring(doc, encoding=str)
+
   else:
-    article = article_from_cache(post['id'], post['updated'])
-    if not article:
-      base.STATSC.incr('zhihu.cache_miss')
-      try:
-        _article_q.put_nowait(str(post['id']))
-      except asyncio.QueueFull:
-        logger.warning('_article_q full')
-        base.STATSC.incr('zhihu.queue_full')
-
-      if fullonly:
-        return None
-
-      content = post['excerpt'] + ' (全文尚不可用)'
-      content = zhihulib.process_content_for_html(content, pic=pic)
-
-    else:
-      # logger.debug('cache hit for %s', post['id'])
-      base.STATSC.incr('zhihu.cache_hit')
-
-      content = article['content']
-      if pic:
-        doc = fromstring(content)
-        base.proxify_pic(doc, zhihulib.re_zhihu_img, pic)
-        content = tostring(doc, encoding=str)
+    base.STATSC.incr('zhihu.cache_miss')
+    try:
+      _article_q.put_nowait(str(post['id']))
+    except asyncio.QueueFull:
+      logger.warning('_article_q full')
+      base.STATSC.incr('zhihu.queue_full')
+
+    if fullonly:
+      return None
+
+    content = post['excerpt'] + ' (全文尚不可用)'
+    content = zhihulib.process_content_for_html(content, pic=pic)
 
   if post.get('title_image'):
     content = '<p><img src="%s"></p>' % post['title_image'] + content
 
-  item = PyRSS2Gen.RSSItem(
+  return PyRSS2Gen.RSSItem(
```

Comment:
The two layers of `if` checks distinct aspects of things and should not be merged.

Swapping the `if` is undesired because when you get data from cache, you first think what if it's not there before continuing.